scripts/safari.js

```diff
@@ -7568,287 +7568,280 @@ function Safari() {
         safaribot.sendMessage(src, "Winners will receive " + plural(this.amount, this.reward.name) + "!", safchan);
     };
     
-    function Marathon(src, reward, amount) {
+    function PokeRace(src, minBet, maxBet, favorite, underdog, normal, goal, silver) {
         SafariEvent.call(this);
-        this.eventName = "Pokémon Marathon";
+        this.eventName = "Pokémon Race";
         
-        this.amount = reward.type == "item" ? amount : 1;
-        this.reward = reward;
-        this.hasReward = true;
+        this.bets = {};
+        this.runners = {};
         
-        this.joinmsg = "Type " + link("/signup") + " "  + " to participate! Winners will receive <b>" + plural(amount, reward.name) + "</b>!";
+        this.turnLength = 5;
+        this.signupsDuration = 6;
+        this.minPlayers = 1;
         
-        this.goal = 1000;
-        this.turnLength = 7;
-        this.minPlayers = 4;
-        this.maxWinners = 1;
-        this.runners = [];
-        this.attackFrequency = {};
+        this.silver = silver || false;
+        this.underdogPay = underdog || 2;
+        this.favoritePay = favorite || 1.25;
+        this.normalPay = normal || 1.75;
         
-        this.eventCommands = {
-            attack: this.attack
-        };
+        this.minBet = minBet || 20;
+        this.maxBet = maxBet || 500;
+        this.goal = goal || 30;
+        this.betList = [];
+        
+        var r;
+        while (Object.keys(this.runners).length < 6) {
+            r = sys.pokemon(sys.rand(1, 722));
+            if (!(r in this.runners)) {
+                this.runners[r] = 0;
+            }
+        }
+        this.favorite = Object.keys(this.runners).random();
+        do {
+            this.underdog = Object.keys(this.runners).random();
+        } while (this.underdog == this.favorite);
+        
+        this.racersList = Object.keys(this.runners).map(function(x) {
+            if (x == this.favorite) {
+                return "<font color='red'>" + x + "</font> (Favorite)";
+            } else if (x == this.underdog) {
+                return "<font color='red'>" + x + "</font> (Underdog)";
+            } else {
+                return "<font color='red'>" + x + "</font>";
+            }
+        }, this);
+        this.icons = {};
+        for (var e in this.runners) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.